app/routes/auth/github.tsx

```diff
@@ -1,19 +1,33 @@
-import { ActionFunction } from "@remix-run/node";
+import { ActionFunction, redirect } from "@remix-run/node";
 import config from "~/config";
 import { authenticator } from "~/services/auth.server";
 import { ensureUserCookie } from "~/shared/session";
+import { LOGIN_ERROR_CODES } from "~/shared/session/useLoginErrors";
 
 export default function GH() {
   return null;
 }
 
 export const action: ActionFunction = async ({ request }) => {
-  const { userId } = await ensureUserCookie(request);
-  return authenticator.authenticate("github", request, {
-    context: {
-      userId,
-    },
-    successRedirect: config.dashboardPath,
-    failureRedirect: config.loginPath,
-  });
+  try {
+    const { userId } = await ensureUserCookie(request);
+    return await authenticator.authenticate("github", request, {
+      context: {
+        userId,
+      },
+      successRedirect: config.dashboardPath,
+      throwOnError: true,
+    });
+  } catch (error: unknown) {
+    // all redirects are basically errors and in that case we don't want to catch it
+    if (error instanceof Response) return error;
+    if (error instanceof Error) {
+      console.log(error);
```

Comment:
@todo

Reply:
I wonder if we should start logging to sentry right away

Reply:
we definitely shouldn't use console.* api because we need to lint it to prevent keeping console.* left overs from debugging by accident, so we will need to use a different logging functions generally, but maybe we should just start using sentry

Reply:
Do we have sentry?

Reply:
nope, but let me create it in a sec

Reply:
done, you should have received an invite

Reply:
one important thing is to make sure we can distinguish logs in sentry from production and always pass userid,projected etc, whatever is available to reproduce the error later

Reply:
check it out, if adding sentry is not trivial, we can  make a separate PR

Reply:
specifically we need to make sure that sentry is using fetch API to send the events, not node apis, because otherwise that won't work in a CF worker

Reply:
oh there is a client for sentry that works in CF worker, I bet it works in node too, but need to check https://github.com/robertcepa/toucan-js

Reply:
Made a new PR for sentry